### PR TITLE
Move some network code to a libary shared between chef and crowbar

### PR DESCRIPTION
**Pull request by rhafer**
There a still some missing pieces

- [x] add network-json-resolv cli tool
- [x] adjust test-cases chef/cookbooks/barclamp/libraries/test.rb
- [x] testing with some different networking setups

### CODE REVIEW

**Comment by rhafer** [comment]
The currently gating failure should be fixed by: https://github.com/crowbar/crowbar-openstack/pull/532

**Comment by vuntz** [comment]
LGTM so far.

**Comment by toabctl** [comment]
+1

**Comment by rhafer** [comment]
rebased the commits and addressed the last couple of issues:

* test.rb is working now
* network-json-resolve util is added (basically  unchanged from @vuntz original version)
* did a few more tests (especially with a bonding setup)

**Comment by rhafer** [comment]
@toabctl would you mind taking another look?

File: bin/network-json-resolve [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+  def cr_network_config
+    @network_json
+  end
+
+  def cr_node_roles
+    (@node.roles + @roles).uniq
+  end
+
+  def cr_error(s)
+    # nothing
```
Comment (toabctl) [inline bin/network-json-resolve]: Shouldn't that be a ```puts s``` ? But can be done later I guess.

Comment (rhafer) [inline bin/network-json-resolve]: fixed

**Comment by rhafer** [comment]
> One commit msg has a typo "This
> seems to work find" (fine instead of find). But +1

The gating failed anyway. So I fixed the typo as well.

**Comment by rhafer** [comment]
Hm, now hound is running amok. Complaining about lines longer than 80. IIRC we set that to 100 on all the crowbar repos.

**Comment by vuntz** [comment]
(not merging yet due to the freeze)

File: bin/network-json-resolve [hunk 2]
```diff
@@ -0,0 +1,394 @@
+#!/usr/bin/env ruby
+#
+# Copyright 2014-2016, SUSE LINUX Products GmbH
```
Comment (dirkmueller) [inline bin/network-json-resolve]: We're SUSE Linux GmbH nowadays ;)

File: crowbar_framework/lib/crowbar/conduit_resolver.rb [hunk 3]
```diff
@@ -0,0 +1,328 @@
+#
+# Copyright 2011-2013, Dell
+# Copyright 2014, SUSE LINUX Products GmbH
```
Comment (dirkmueller) [inline crowbar_framework/lib/crowbar/conduit_resolver.rb]: SUSE Linux GmbH

**Comment by rhafer** [comment]
Force pushed again to adjust the Copyright for the lib and network-json-resolve

**Pull request merged**